tests/org.eclipse.n4js.jsdoc.tests/src/org/eclipse/n4js/jsdoc/MD2HTMLConvertingBuilderTest.xtend

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.jsdoc
+
+import org.junit.Assert
+import org.junit.Test
+
+class MD2HTMLConvertingBuilderTest {
+	
+	
+	def assertConvert(CharSequence markdown, CharSequence expectedHTML) {
+		val html = MD2HTMLConvertingBuilder.convert(markdown.toString);
+		Assert.assertEquals(expectedHTML.toString, html);
+	}
+	
+	
+	@Test
+	def testEmpty() {
+		assertConvert("", "");	
+	}
+	
+	@Test
+	def testSingleLine() {
+		assertConvert("Single line.", "Single line.");	
+	}
+	
+	@Test
+	def testTwoLines() {
+		assertConvert(
+			'''
+			First line.
+			Second line.
+			''',
+			'''
+			First line.
+			Second line.
+			''');	
+	}
+
+	@Test
+	def testTwoParagraphs() {
+		assertConvert(
+			'''
+			First par.
+			
+			Second par.
+			''',
+			'''
+			First par.
+			<p>Second par.
+			''');	
+	}
+	
+	@Test
+	def testList() {
+		assertConvert(
+			'''
+			This is a list:
+			- hello
+			- world
+			''',
+			'''
+			This is a list:
+			<ul>
+			<li> hello
+			<li> world
+			</ul>
+			''');	
+	}
+	
+	@Test
+	def testMultiLineList() {
+		assertConvert(
+			'''
+			This is a list:
+			- hello
+			  hallo
```

Comment:
"hallo" is displayed in a new line under "hello". Long entries that span across multiple lines are not possible if comments break at character max. count (e.g. 120).

BTW: The "\<li>" you use here is actually "\<li/>"

Reply:
<li> without </li> is not <li/>. 

"<li/>xxxx"  is similar to "<li></li>xxxx"

<li> and <p> are automatically closed by most browsers.

I didn't want to close it to not complicate the code too much. And the hover can handle it.

Reply:
Markdown does not need empty lines.
- list item
  - nested item
Back to text

Back to text

Reply:
Ah -- as shown in the previous comment, markdown needs an empty line to end the list. Will fix that.